examples/angular_bazel_architect/angular.json

```diff
@@ -17,6 +17,7 @@
         "build": {
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
+            "forkTypeChecker": false,
```

Comment:
Ahhh nice one on finding the cause of this issue.  
Could you try enabling windows ci now that this works also?

Reply:
Enabled, let's wait and see :)

Reply:
There seems to be the same error for `jest-workers`

```
Error: spawn C:\b\ojvxx6nx\execroot\build_bazel_rules_nodejs\_tmp\00ec2cae7910fb9a26b84ca52e1907d7\_bazel_b\r453ukw2\external\build_bazel_rules_nodejs\internal\node\_node_bin\node ENOENT
--
  | at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
  | at onErrorNT (internal/child_process.js:456:16)
  | at processTicksAndRejections (internal/process/task_queues.js:80:21)
  | Emitted 'error' event on ChildProcess instance at:
  | at Process.ChildProcess._handle.onexit (internal/child_process.js:270:12)
  | at onErrorNT (internal/child_process.js:456:16)
  | at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  | errno: 'ENOENT',
  | code: 'ENOENT',
  | syscall: 'spawn C:\\b\\ojvxx6nx\\execroot\\build_bazel_rules_nodejs\\_tmp\\00ec2cae7910fb9a26b84ca52e1907d7\\_bazel_b\\r453ukw2\\external\\build_bazel_rules_nodejs\\internal\\node\\_node_bin\\node',
  | path: 'C:\\b\\ojvxx6nx\\execroot\\build_bazel_rules_nodejs\\_tmp\\00ec2cae7910fb9a26b84ca52e1907d7\\_bazel_b\\r453ukw2\\external\\build_bazel_rules_nodejs\\internal\\node\\_node_bin\\node',
  | spawnargs: [
  | '--require',
  | 'C:/b/ojvxx6nx/execroot/build_bazel_rules_nodejs/_tmp/00ec2cae7910fb9a26b84ca52e1907d7/_bazel_b/r453ukw2/external/build_bazel_rules_nodejs/internal/node/node_patches.js',
  | '--preserve-symlinks',
  | '--preserve-symlinks-main',
  | 'C:\\b\\ojvxx6nx\\execroot\\build_bazel_rules_nodejs\\_tmp\\00ec2cae7910fb9a26b84ca52e1907d7\\_bazel_b\\r453ukw2\\execroot\\angular_bazel_architect\\node_modules\\jest-worker\\build\\workers\\processChild.js'
  | ]
  | }
```

I am thinking this might be related to something how the actual spawn of the architect is happening from bazel. This will need some digging.

Reply:
JFYI: https://github.com/angular/angular-cli/issues/16119

Reply:
This is the code that puts the bazel node into the PATH https://github.com/bazelbuild/rules_nodejs/blob/master/internal/node/require_patch.js#L30
but that should be fine

Reply:
Most likely the problem is in this line https://github.com/bazelbuild/rules_nodejs/blob/25600ea4c307c120d04a14c6d41644a3e8d3ed3a/packages/node-patches/src/subprocess.ts#L50

That should probably have an extension of `exe` or  `bat` for windows.

Unfortunately the `node-patches` tests are all disabled for windows. 

//cc @gregmagolan & @alexeagle

Reply:
Changing it to `bat`  for windows does work, but eventually it will fail due `bad file descriptor, uv_pipe_open` in `jest-workers` when trying to fork the process.

That is because on Windows `bat` and `cmd` are non executable, on their own without the `shell` option defined. 

So, changing the above line to ` process.argv[0] = process.execPath` yield the correct result in Windows, though of course it's not setting the patched env variables which are configured in the batch file.

Is there any reason why we don't set `process.env` instead since forked/spawned processes should inherit the env vars from the parent process.

Reply:
Jest-workers do offer the possibility to use pass additional option to the fork process which in this case we can tell you use `shell`. But this would be a "workaround", as other libs that uses forked processes might not offer this functionality.

Reply:
@soldair could chime in about the node-patches issue

Reply:
we either need a different executable that works without shell or we can force the shell option to true on windows if the executable is node. Seems easiest to do the last one

Reply:
if someone can make a really smol .exe to use instead of the bat that should work also but i'm less clear on how to do that.

Reply:
@soldair how can we force the shell open on processes forked in 3rd party libraries?

Reply:
Like node-patches modifies the global fs object, we could patch the child_process object such that when the executable  to spawn is the batch file we set shell:true and we make execFile do similar trick. Much like baking in this workaround for execFile. https://nodejs.org/api/child_process.html#child_process_spawning_bat_and_cmd_files_on_windows

The alternative is to build the shim into a proper `.exe` file which @alexeagle mentioned and we don't have to modify child_process at the cost of whatever we need to do to build the exe. Right now it's dynamically created when you run node the first time.